docs/Generators/Top-Testharness.rst

```diff
@@ -0,0 +1,61 @@
+Tops, TestHarnesses, and TestDriver
+====================================
+
+The three highest levels of hierarchy in a Chipyard
+SoC are the Top, TestHarness, and the Testbench. The Top and TestHarness are both emitted by Chisel generators.
+
+Top/DUT
+-------------------------
+The top-level module of a Rocketchip SoC is composed via cake-pattern. Specifically, "Tops" extend a "System", which extends a "Subsystem", which extends a "BaseSubsystem"
+
+
+BaseSubsystem
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The BaseSubsystem is defined in ``generators/rocketchip/src/main/scala/subsystem/BaseSubsystem.scala``. Looking at the BaseSubsystem abstract class, we see that this class instantiates the top-level buses (frontbus, systembus, peripherybus, etc.), but does not specify a topology. We also see this class define several "ElaborationArtefacts", for example the device tree string, and the diplomacy GraphML.
```

Comment:
Can you make this a link to the code also?

Reply:
I would also make BaseSubsystem be a code word.

Reply:
What do you mean "link to the code". Like a hyperlink?

Reply:
I think RocketChip structure will change soon so this link will be dead.